public/index.js

```diff
@@ -236,6 +240,20 @@
       });
     }
 
+    function addHaloClickEvent(element){
```

Comment:
Okay, I see that addHalo is defined in and requires inner scope.

Generally this file is getting difficult to review.. :(

We *need* to manage the state in this file. For now I can let this go. But we need to fix this file..
Message me on gitter and we will work out if you have time in this sprint